partiql-spi/src/main/kotlin/org/partiql/spi/connector/ConnectorObjectHandle.kt

```diff
@@ -15,9 +15,12 @@
 package org.partiql.spi.connector
 
 /**
- * Holds a PartiQL Value's representation within a Catalog and its location within the Catalog.
+ * Holds a  Catalog and its location within the Catalog.
+ *
+ * @property path   The absolute path to this object in the catalog
+ * @property obj    Arbitrary metadata associated with this object.
  */
 public class ConnectorObjectHandle(
-    public val absolutePath: ConnectorObjectPath,
-    public val value: ConnectorObject
+    public val path: ConnectorPath,
+    public val obj: ConnectorObject,
```

Comment:
What is the need of a separate `ConnectorObject` interface, if this is just going to be holding metadata.

Reply:
ConnectorObject is for arbitrary metadata about a catalog's values. ConnectorFunctionHandle is only for function metadata

Reply:
I see your question is actually slightly different.

ConnectorObject is not analogous to ConnectorFunction. ConnectorObject would be better named as ConnectorObjectMetadata. It is most similar to FunctionSignature, but can be anything.